torch/csrc/jit/codegen/cuda/index_idgraph_utils.cpp

```diff
@@ -116,6 +154,80 @@ IndexingParameters getGlobalIndexParameters(
   return index_parameters;
 }
 
+// Initial index parameters for shared and local case
+IndexingParameters getNonGlobalInitialIndexParameters(
+    const LoopIndexing& loop_indexing,
+    const TensorView* consumer_tv,
+    bool index_producer = false,
+    const TensorView* producer_tv = nullptr,
+    std::unordered_map<IterDomain*, IterDomain*> p2c_map = {}) {
+  IndexingParameters index_parameters;
+  const auto& loops = loop_indexing.loops();
+  const auto& loop_domains = loop_indexing.loopDomains();
+
+  // TODO:
+  //  The non-global path should become shorter as we
+  // pull more info into id graph.
+  std::unordered_map<IterDomain*, IterDomain*> alloc_id_map;
+
+  if (index_producer) {
+    alloc_id_map = mapAllProducerDomainsToConsumer(producer_tv, consumer_tv);
+  }
+
+  auto alloc_tv = index_producer ? producer_tv : consumer_tv;
+  auto alloc_info = loop_utils::getAllocInformation(
+      alloc_tv, loops, alloc_id_map, index_producer);
+
+  std::unordered_map<kir::ForLoop*, Val*> loop_to_ind_map;
+  std::unordered_set<kir::ForLoop*> zero_loops;
+
+  kir::ForLoop* double_buffer_loop = nullptr;
+
+  if (index_producer) {
+    double_buffer_loop =
+        GpuLower::current()->doubleBufferInfo().getDoubleBufferLoop(
+            consumer_tv, loops, true);
+  }
+
+  std::tie(loop_to_ind_map, zero_loops) = indexMapFromTV(
+      alloc_tv,
+      loops,
+      alloc_info.init_for_loop,
+      !index_producer,
+      double_buffer_loop);
+
+  ensureStaticIndexing(alloc_tv, alloc_info.init_for_loop, loops, alloc_id_map);
```

Comment:
Not a note for this PR, but generally a note: can't we do this pass much earlier outside of indexing? CC @naoyam